### ISSUE TITLE
Bug? Node v0.11 and modules/checker checkStdin test

### ISSUE THREAD

**Issue by zxqfox** [jscs-dev/node-jscs#869]
I've used node 0.11 via nvm, ran mocha, and saw:
```
  1 failing

  1) modules/checker checkStdin returns a promise:
     TypeError: Attempted to wrap on which is already wrapped
      at Object.wrapMethod (/home/alex/repos/node-jscs/node_modules/sinon/lib/sinon/util/core.js:78:25)
      at Object.spy (/home/alex/repos/node-jscs/node_modules/sinon/lib/sinon/spy.js:34:26)
      at Context.<anonymous> (/home/alex/repos/node-jscs/test/checker.js:62:29)
      at callFn (/home/alex/repos/node-jscs/node_modules/mocha/lib/runnable.js:251:21)
      at Test.Runnable.run (/home/alex/repos/node-jscs/node_modules/mocha/lib/runnable.js:244:7)
      at Runner.runTest (/home/alex/repos/node-jscs/node_modules/mocha/lib/runner.js:374:10)
      at /home/alex/repos/node-jscs/node_modules/mocha/lib/runner.js:452:12
      at next (/home/alex/repos/node-jscs/node_modules/mocha/lib/runner.js:299:14)
      at /home/alex/repos/node-jscs/node_modules/mocha/lib/runner.js:309:7
      at next (/home/alex/repos/node-jscs/node_modules/mocha/lib/runner.js:248:23)
      at Immediate._onImmediate (/home/alex/repos/node-jscs/node_modules/mocha/lib/runner.js:276:5)
      at processImmediate [as _immediateCallback] (timers.js:374:17)
  --------------
  Error: Stack Trace for original
      at Object.wrapMethod (/home/alex/repos/node-jscs/node_modules/sinon/lib/sinon/util/core.js:98:34)
      at Object.spy (/home/alex/repos/node-jscs/node_modules/sinon/lib/sinon/spy.js:34:26)
      at Context.<anonymous> (/home/alex/repos/node-jscs/test/checker.js:54:29)
      at callFn (/home/alex/repos/node-jscs/node_modules/mocha/lib/runnable.js:251:21)
      at Test.Runnable.run (/home/alex/repos/node-jscs/node_modules/mocha/lib/runnable.js:244:7)
      at Runner.runTest (/home/alex/repos/node-jscs/node_modules/mocha/lib/runner.js:374:10)
      at /home/alex/repos/node-jscs/node_modules/mocha/lib/runner.js:452:12
      at next (/home/alex/repos/node-jscs/node_modules/mocha/lib/runner.js:299:14)
      at /home/alex/repos/node-jscs/node_modules/mocha/lib/runner.js:309:7
      at next (/home/alex/repos/node-jscs/node_modules/mocha/lib/runner.js:248:23)
      at Immediate._onImmediate (/home/alex/repos/node-jscs/node_modules/mocha/lib/runner.js:276:5)
      at processImmediate [as _immediateCallback] (timers.js:374:17)
```

/cc @mrjoelkemp

**Comment by mrjoelkemp** [comment]
Thanks for the report! Seems like it's specific to your node version. I had the same thing happen some time ago with an older version of node. Travis doesn't like your fix likely because it's running a newer stable version of node. 

Try upgrading locally if you can? 

Happy new year,  friend :)

**Comment by zxqfox** [comment]
@mrjoelkemp oooooo... Black magic detected.

Can you take a look at [Spying on existing methods](http://sinonjs.org/docs/)?

`sinon.spy` rewriting global object's method so it MUST restore it after calls counting (if it doesn't silently). I'm pretty sure in it. And 0.11.14 doesn't do this for sure. I'm not sure about 0.10.35.

v0.10.35 really fails with this fix while v0.11.14 starting works well ;-(

**Comment by zxqfox** [comment]
@mrjoelkemp :christmas_tree: I think I've found working solution for both nodes: `spy.restore()`.
